packages/core/src/new-api/internal/utils/retrieve-nested-runtime-values.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  isFuture,
+  isModuleParameterRuntimeValue,
+  isRuntimeValue,
+} from "../../type-guards";
+import {
+  ArgumentType,
+  ModuleParameterRuntimeValue,
+  RuntimeValueType,
+} from "../../types/module";
+
+export function retrieveNestedRuntimeValues(
+  args: ArgumentType[]
+): Array<ModuleParameterRuntimeValue<any>> {
+  return args.flatMap(checkForValues).filter(isModuleParameterRuntimeValue);
+}
+
+function checkForValues(
+  arg: ArgumentType
+):
+  | Array<ModuleParameterRuntimeValue<any> | null>
+  | ModuleParameterRuntimeValue<any>
+  | null {
+  if (isRuntimeValue(arg)) {
+    if (arg.type === RuntimeValueType.ACCOUNT) {
+      return null;
```

Comment:
Doesn't this mean that we aren't validating that the account's index exist? We should validate params and account runtime values, as both can be missing for different reasons.

Reply:
This is for validating module parameters, and account runtime values can't be passed in as a module parameter afaik. This specific line is just type narrowing.